model_discretization_api.py

```diff
@@ -0,0 +1,209 @@
+# models have to be integrated over pixels, for this purpose models could define
+# an integrate function, that takes upper and lower bounds of the integration
+# (see also https://github.com/astropy/astropy/issues/1586)
+model = Gaussian1D(1, 0, 1)
+model.integrate(0, np.inf)
+
+# it should also work with arrays of bounds of course
+x_lo = np.arange(-10, 11) - 0.5
+x_hi = np.arange(-10, 11) + 0.5
+model.integrate((x_lo, x_hi))
+
+# for 2d models it would be
+y_lo = np.arange(-10, 11) - 0.5
+y_hi = np.arange(-10, 11) + 0.5
+model.integrate((x_lo, x_hi, y_lo, y_hi))
+
+# whenever possible (e.g. polynomials) the integrate function would be implemented
+# analytically, for all others a standard numerical integration routine would be used
+# e.g. from scipy.integrate
+# the integration mode can be passed to the integrate method if it is not already an 
+# attribute of model
+model.integrate((x_lo, x_hi, y_lo, y_hi), mode='integrate')
+
+# for convenience one could incorporate the functionality into the standard model API
+# as is done for the convolution kernals 
+model = Gaussian1D(1, 0, 1, mode='center') 
+model = Gaussian1D(1, 0, 1, mode='linear_interp') 
+model = Gaussian1D(1, 0, 1, mode='integrate')
+model = Gaussian1D(1, 0, 1, mode='oversample', factor=10)
+
+# the given mode could also be used when the model is called for evaluation
+x = np.arange(-10, 11)
+y = np.arange(-10, 11)
+model(np.meshgrid(x,y))
+
+
+# models that are spatially confined to a certain region such as Gaussian2D, Disk2D,
+# Delta2D, Box2D, ... should have the possibility to apply a different normalization,
+# where the amplitude parameter corresponds to the integral of the model and not the
+# peak value at maximum (see e.g.: https://gammapy.readthedocs.org/en/latest/api/gammapy.morphology.Shell2D.html#gammapy.morphology.Shell2D)
+# This is very useful for modeling sources, because the amplitude parameter then
+# corresponds to the total flux of the source...
+model = Gaussian2D(1, 0, 0, 1, 1, normalization='peak')
+model = Gaussian2D(1, 0, 0, 1, 1, normalization='integral')
+
+# alternatively one could introduce new NormGaussian2D, NormBox2D, ... models, but that
+# would probably be a lot of code duplication?
+
+# all models that are 'normalisable' should have an `extent` attribute that specifies
+# a rectangular region where the model is nonzero or has 99.99% containment (or a similar
+# criterion). A reasonable value would be chosen as default and would be given as multiples 
+# model's width parameters. Model evaluation can later be limited to the region defined
+# by `extent` for better performance 
+Gaussian2D.extent= dict(x=5 * x_stddev, y=5 * y_stddev)
+Disk2D.extent = dict(x=R_0, y=R_0) 
+
+# models should have an attribute that defines a reasonable sample size for the model,
+# warnings could be raised, when models are undersampled.
+Gaussian2D.sample_size = width / 5.
+
+
+###############################
+# Model Image Computation API #
```

Comment:
@eteq – What exactly are you proposing here?
If the model evaluate returns a quantity, it'll just be used, i.e. the output sky image will also be a quantity.
Other than that, I don't think we should introduce checks or attach units ourselves, because it would be restricting.
Maybe Joe wants to render an image in flux units and Jane in counts units and Jim in surface brightness units?